blacklists.py

```diff
@@ -140,17 +147,174 @@ def exists(self, item: Union[str, dict]):
                 splat = x.split('\t')
                 if len(splat) == 3 and splat[2].strip() == item:
                     return True, i
+        return False, -1
+
+
+class YAMLParserCIDR(BlacklistParser):
+    """
+    YAML parser for IP blacklist (name suggests we should move to proper CIDR eventually).
+
+    Base class for parsers for YAML files with simple schema validation.
+    """
+    # Remember to update the schema version if any of this needs to be changed
+    SCHEMA_VERSION = '2019120601'  # yyyy mm dd id
+    SCHEMA_VARIANT = 'yaml_cidr'
+    SCHEMA_PRIKEY = 'ip'
+
+    def __init__(self, filename):
+        super().__init__(filename)
+
+    def _parse(self, keep_disabled=False):
+        with open(self._filename, 'r', encoding='utf-8') as f:
+            y = yaml.safe_load(f)
+        assert y['Schema'] == self.SCHEMA_VARIANT
```

Comment:
Rather than asserts, can these be `if` checks with a `log('error', '...')` call if they fail? Logging then exiting is preferable to just crashing.

Reply:
I guess that should be changed; I see the TSV parser has a `try`/`except` around similar constructs. I suppose I could use that here too. I was thinking perhaps there should be a stricter validation in the unit tests but I haven't worked on that yet.

Reply:
I updated to replace the asserts, too, with raise ValueError. I'm still thinking about how to robustly trap the errors, and whether in fact it makes more sense to trap and continue. These are integrity errors which in some sense _should_ fail, though of course it's nice if it doesn't ruin the other parts of Smokey even when there is a problem.

Reply:
I added a commit which allows Smokey to start up even if the blacklists are hosed. It will still fail in interesting ways if you attempt to add or remove something invalid; but doing that programmatically is going to be in the scope of a separate PR, so I'm not touching the `gitmanager.py` code yet.

Reply:
I need to look at the new commits to see what the update was, but just based on the text description:

I would expect SD failing to start up when the blacklists are hosed would be a feature. I'm not sure we want to enable it to properly start when the blacklists are messed up, unless that's behind an option of some sort. Blacklist integrity is something we want to maintain and we want SD to be able to use the data in the external lists.

As I understand it, currently, if any of the blacklists fail integrity checks, then SD will fail to start. If SD fails to start three times in a row, then nocrash.py will move the instance to the next oldest commit. This process repeats until it finds a commit which runs (e.g. where the blacklists are not messed up). Are we sure we want to do something other than that? Why would we want to run, in production, on a commit where the blacklists are messed up?

Reply:
So I will just drop that commit for now then.